pkg/validation/spec/ref.go

```diff
@@ -149,19 +151,12 @@ func (r *Ref) UnmarshalJSON(d []byte) error {
 }
 
 func (r *Ref) fromMap(v map[string]interface{}) error {
-	if v == nil {
-		return nil
+	jsonRef, err := internal.JSONRefFromMap(v)
+	if err != nil {
+		return err
 	}
-
-	if vv, ok := v["$ref"]; ok {
-		if str, ok := vv.(string); ok {
-			ref, err := jsonreference.New(str)
-			if err != nil {
-				return err
-			}
-			*r = Ref{Ref: ref}
-		}
+	if jsonRef != nil {
+		*r = Ref{Ref: *jsonRef}
```

Comment:
Can't we just set it, even if the ref is empty? That's also more consistent with what we were doing?
```suggestion
	r.Ref = jsonRef
```

Reply:
We're only setting the ref `if vv, ok := v["$ref"]; ok {` previously, so this behavior is consistent (although I also dislike that we're returning a pointer). We could make it semi-consistent by returning an empty jsonRef and dropping the pointer, resulting in an extra memory alloc (probably negligible?) every time the ref should be nil.

Reply:
Yeah, and the merge logic wants us to not unset it if it's unset. What about passing a pointer to the ref, like this:
```golang
err := internal.JSONRefFromMap(&r.Ref, v)
```

Reply:
We would have a circular import reference between `validation/spec` and `internal` so we can't expose `spec.Ref` to the internal package.

Reply:
Thanks for the suggestion, this looks much cleaner!